From bbff99fa2a0a044710673c6ebbf75cf55fbf12ae Mon Sep 17 00:00:00 2001
From: Johannes Schindelin <johannes.schindelin@gmx.de>
Date: Wed, 24 Jan 2018 10:57:29 +0100
Subject: [PATCH] Work around LN_S="ln"

It is possible to link a directory symbolically into a subdirectory of
itself. But it is not possible to hard-link it.

Since MSYS2 does not really support symbolic links out of the box (and
tries hard-links or even copies instead), that would break the build.

Signed-off-by: Johannes Schindelin <johannes.schindelin@gmx.de>
---
 lib/hcrypto/Makefile.am | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/lib/hcrypto/Makefile.am b/lib/hcrypto/Makefile.am
index 469176b..e95345b 100644
--- a/lib/hcrypto/Makefile.am
+++ b/lib/hcrypto/Makefile.am
@@ -9,7 +9,7 @@ AM_CPPFLAGS += $(INCLUDE_openssl_crypto)
 endif
 
 AM_CPPFLAGS += -I$(top_srcdir)/lib/hx509 \
-	       -I$(srcdir)/libtommath -DUSE_HCRYPTO_LTM=1
+	       -I$(srcdir)/libtommath -I.. -DUSE_HCRYPTO_LTM=1
 
 lib_LTLIBRARIES = libhcrypto.la
 check_LTLIBRARIES = libhctest.la
@@ -308,6 +308,7 @@ $(libhcrypto_la_OBJECTS): $(srcdir)/version-script.map
 
 
 hcrypto-link:
+	test ln = "$(LN_S)" || \
 	$(LN_S) $(srcdir)/../hcrypto hcrypto
 	touch hcrypto-link
 
-- 
2.16.1.windows.1

